### PR TITLE
[Rule Tuning] NTDS or SAM Database File Copied

### LINKED ISSUE #1376: [Rule Tuning] NTDS or SAM Database File Copied
## Link to rule
https://github.com/elastic/detection-rules/blob/main/rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml

## Description

**Current Query**

```
process where event.type in ("start", "process_started") and
 process.pe.original_file_name in ("Cmd.Exe", "PowerShell.EXE", "XCOPY.EXE") and
 process.args : ("copy", "xcopy", "Copy-Item", "move", "cp", "mv") and
 process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*")
```

**Proposed Query**

Reference Leading to Change:
- https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1003.002/T1003.002.md#atomic-test-3---esentutlexe-sam-copy

```
process where event.type in ("start", "process_started") and
 process.pe.original_file_name in ("Cmd.Exe", "PowerShell.EXE", "XCOPY.EXE", "esentutl.exe") and
 process.args : ("copy", "xcopy", "Copy-Item", "move", "cp", "mv","*/y*", "*/vss*", "*/d*") and
 process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*","*/system32/config/SAM*", "*/SAM*")
```

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1376

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by austinsonger** [comment]
@threat-punter I'm thinking about expanding this.

```
process where event.type in ("start", "process_started") and
 process.pe.original_file_name in ("Cmd.Exe", "PowerShell.EXE", "XCOPY.EXE", "esentutl.exe") and
 process.args : ("copy", "xcopy", "Copy-Item", "move", "cp", "mv","*/y*", "*/vss*", "*/d*") and
 process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*","*/system32/config/SAM*", "*/SAM*","*\\config\\security*","\\config\\system","*\\repair\\sam*","*\\repair\\system*","\\repair\\security*","*\\config\\RegBack\\system*","*\\config\\RegBack\\security*")
```

- I know `\\config\\RegBack\\` isn't part of the newest versions of Windows 10, but you would be surprised how many people  don't update and still have Windows 7.

I think a name change would be in place to something like

> **Sensitive File or Database File Copied**

File: rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml [hunk 1]
```diff
@@ -25,9 +28,9 @@ type = "eql"
 
 query = '''
 process where event.type in ("start", "process_started") and
- process.pe.original_file_name in ("Cmd.Exe", "PowerShell.EXE", "XCOPY.EXE") and
- process.args : ("copy", "xcopy", "Copy-Item", "move", "cp", "mv") and
- process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*")
+ process.pe.original_file_name in ("Cmd.Exe", "PowerShell.EXE", "XCOPY.EXE", "esentutl.exe") and
+ process.args : ("copy", "xcopy", "Copy-Item", "move", "cp", "mv","*/y*", "*/vss*", "*/d*") and
```
Comment (threat-punter) [inline rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml:32]:
```suggestion
 process.args : ("copy", "xcopy", "Copy-Item", "move", "cp", "mv", "*/y*", "*/vss*", "*/d*") and
```

File: rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml [hunk 2]
```diff
@@ -25,9 +28,9 @@ type = "eql"
...
+ process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*","*/system32/config/SAM*", "*/SAM*")
```
Comment (threat-punter) [inline rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml:33]:
```suggestion
 process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*", "*/system32/config/SAM*", "*/SAM*")
```

**Review by threat-punter — COMMENTED** [review]
@austinsonger thanks for opening this PR. I'm open to your suggestion. Do you think it makes sense to expand this rule or create an additional one?

**Comment by austinsonger** [comment]
I think it being included in this rule is good enough.

**Review by threat-punter — APPROVED** [review]

File: rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml [hunk 3]
```diff
@@ -25,9 +28,9 @@ type = "eql"
 
 query = '''
 process where event.type in ("start", "process_started") and
- process.pe.original_file_name in ("Cmd.Exe", "PowerShell.EXE", "XCOPY.EXE") and
- process.args : ("copy", "xcopy", "Copy-Item", "move", "cp", "mv") and
- process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*")
+ process.pe.original_file_name in ("Cmd.Exe", "PowerShell.EXE", "XCOPY.EXE", "esentutl.exe") and
+ process.args : ("copy", "xcopy", "Copy-Item", "move", "cp", "mv", "*/y*", "*/vss*", "*/d*") and
+ process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*", "*/system32/config/SAM*", "*/SAM*")
```
Comment (brokensound77) [inline rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml:33]: We should keep the associated args and processes  separate to be more specific with the logic. Even though someone running them may be nonsensical (like `esentutil.exe copy-item ...`) it is still good practice to make the logic specific and obvious for readability down the road.

```suggestion
  (
    (process.pe.original_file_name in ("Cmd.Exe", "PowerShell.EXE", "XCOPY.EXE") and
       process.args : ("copy", "xcopy", "Copy-Item", "move", "cp", "mv")
    ) or
    (process.pe.original_file_name : "esentutl.exe" and process.args : ("*/y*", "*/vss*", "*/d*"))
  ) and
  process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*", "*/system32/config/SAM*", "*/SAM*")
```

Comment (brokensound77) [reply rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml:33]: Any way to tighten down `"*/SAM*"`?

Also, that would include `"*/system32/config/SAM*"`, making it redundant

Comment (austinsonger) [reply rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml]: Yeah  agree.

**Review by brokensound77 — APPROVED** [review]
LGTM, thanks 👍

File: rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml [hunk 4]
```diff
@@ -25,9 +28,13 @@ type = "eql"
...
- process.pe.original_file_name in ("Cmd.Exe", "PowerShell.EXE", "XCOPY.EXE") and
- process.args : ("copy", "xcopy", "Copy-Item", "move", "cp", "mv") and
- process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*")
+  (
+    (process.pe.original_file_name in ("Cmd.Exe", "PowerShell.EXE", "XCOPY.EXE") and
+       process.args : ("copy", "xcopy", "Copy-Item", "move", "cp", "mv")
+    ) or
+    (process.pe.original_file_name : "esentutl.exe" and process.args : ("*/y*", "*/vss*", "*/d*"))
+  ) and
+  process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*", "*/system32/config/SAM*", "*/SAM*")
```
Comment (brokensound77) [inline rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml:37]:
```suggestion
  process.args : ("*\\ntds.dit", "*\\config\\SAM", "\\*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*\\*", "*/system32/config/SAM*")
```

**Pull request merged**